tidy3d/components/tcad/doping.py

```diff
@@ -56,6 +56,47 @@ def check_bounds(cls, values):
 
         return values
 
+    def _get_indices_in_box(self, coords: dict, meshgrid: bool = True):
```

Comment:
So every geometry (so including `Box`) has an [inside](https://github.com/flexcompute/tidy3d/blob/948098a3940500b7dee71cdefe5bd94600717378/tidy3d/components/geometry/base.py#L89) and [inside_meshgrid](https://github.com/flexcompute/tidy3d/blob/948098a3940500b7dee71cdefe5bd94600717378/tidy3d/components/geometry/base.py#L179) method. Can these be used directly instead of this? It seems like this handles some special cases but wondering if you've considered those.